src/oauth-tooling.ts

```diff
@@ -45,14 +55,21 @@ function createAuthCodeRequestUri(authorizationEndpoint: string, clientId: strin
  * @param authorizationHeaderValue
  * @param accessTokenEndpoint
  * @param realm
+ * @param queryParams optional
  * @returns {Promise<T>|Q.Promise<U>}
  */
 function requestAccessToken(bodyObject: any, authorizationHeaderValue: string,
-                            accessTokenEndpoint: string, realm: string): Promise<Token> {
+                            accessTokenEndpoint: string, realm: string,
+                            queryParams?: {}): Promise<Token> {
 
   const promise = new Promise(function(resolve, reject) {
 
-    fetch(accessTokenEndpoint + '?realm=' + realm, {
+    const queryString = qs.stringify(Object.assign({ realm: realm }, queryParams));
```

Comment:
It is, see examples in https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Object/assign 

"[...]null and undefined will be ignored."